ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -212,6 +214,9 @@ public int compare(IndexEntry e1, IndexEntry e2) {
       if (cleanShutdown) {
         cleanShutdownFile.delete();
       }
+      //Journal should only include the index entry which belongs to the last log segment to make sure it only prevents the last log segment
+      //from being compacted.
+      removeIndexEntryNotBelongsToLastLogSegmentFromJournal();
```

Comment:
we can do this after recovery, so we don't need to have this `inRecovery` variable.